src/Compile-SourceScript/Public/Compile-SourceScript.ps1

```diff
@@ -145,11 +146,48 @@ function Compile-SourceScript {
                     $sourceFile.Name
                 )
             }
+            $returnExitCode = $false
+            if ($MOD_NAME -eq 'sourcemod') {
+                $pluginErrorRegexPattern = '^.*\.sp\(\d+\)\s*:\s*error (\d+)'
+                if (!$PSBoundParameters['SkipWrapper'] -or $env:OS -ne 'Windows_NT') {
+                    $returnExitCode = $true
+                }
+            }elseif ($MOD_NAME -eq 'amxmodx') {
+                $pluginErrorRegexPattern = '^.*\.sma\(\d+\)\s*:\s*error (\d+)'
+                $returnExitCode = $true
+            }
             New-Item -Path $COMPILED_DIR -ItemType Directory -Force | Out-Null
 
             # Begin compilation
             if ($PSBoundParameters['SkipWrapper']) { "Compiling $($sourceFile.Name)..." | Write-Host -ForegroundColor Yellow }
-            Start-Process @processArgs
+
```

Comment:
Would you like to make those changes you have in mind? I see splitting up to be necessary only if the function is getting too polluted. The function `Compile-SourceScript` however does lots of other things that could possibly be moved to separate functions. Whether it's necessary to be splitting up the code is multiple functions is a question, though I think I've previously mentioned how PowerShell cmdlets themselves make it possible to perform multiple steps in a short amount of code. Splitting up code when the same can be achieved via the in-built cmdlets themselves is a common mistake I've made in the past.

Reply:
Yes, the cmdlet is getting very polluted. Better to use a more functional approach, so as to be able to add Pester unit tests which hasn't been implemented yet. I can make the necessary changes

Reply:
The function you mean. I think it has vastly increased in complication due to the use of redirect parameters of `Start-Process` and generation of temporary files. The function was acceptable without the changes from this PR. If we revised to not use redirect output to temporary files as discussed in https://github.com/startersclan/Compile-SourceScript/pull/7#discussion_r468039125, it would save us the hassle of refactoring.

Reply:
A module's main function is called a cmdlet, while it's private functions are called functions. I meant the cmdlet as in `Compile-SourceScript`. 

> The function was acceptable without the changes from this PR. If we revised to not use redirect output to temporary files as discussed in #7 (comment), it would save us the hassle of refactoring.

I'm adding in and you will get what I mean

Reply:
> A module's main function is called a cmdlet, while it's private functions are called functions. I meant the cmdlet as in Compile-SourceScript.

Not true, it depends on whether functions are exported as a cmdlet or not:

- https://github.com/startersclan/Compile-SourceScript/blob/9d4e5de4e9ffd21b58b34580366a3a88d91bc742/build/definitions/modulemanifest/definition.ps1#L26-L29

```
Import-Module .\src\Compile-SourceScript\Compile-SourceScript.psm1 -Force
$ Get-Module Compile-SourceScript

ModuleType Version    Name                                ExportedCommands    
---------- -------    ----                                ----------------    
Script     0.0        Compile-SourceScript                Compile-SourceScript

$ Get-Command Compile-SourceScript

CommandType     Name                                               Version    Source
-----------     ----                                               -------    ------
Function        Compile-SourceScript                               0.0        Compile-SourceScript
```

Anyway, that's besides the point.

Reply:
@joeltimothyoh at least from my understanding it's a common convention that a module's main function(s) should be exported as Cmdlet

Reply:
now that we are using regex patterns as discussed in #7 (comment), there is no more need for this flag (which served to inject a non-zero exit code into `global:LASTEXITCODE` for cases where the compiler binary or compiler wrapper do not return a non-zero exit code even when compilation errors occured). Flag is removed in https://github.com/startersclan/Compile-SourceScript/pull/7/commits/5f4dcebccfd11f2520a3ffefb9caad291815487d